Adding docs for changing server config via the UI & cli (#2210)
kubeshop/tracetest: docs/docs/configuration/trace-polling.md

# Trace Polling Settings

Tracetest currently has one algorithm for gathering the trace associated with a test which is based on periodic polling. We will be adding the ability to add multiple polling settings in the coming releases, as well as adding new polling strategies. Let us know as you have specific needs by [adding an issue](https://github.com/kubeshop/tracetest/issues/new/choose).

## Periodic Polling Strategy

> Tracetest polls the trace data store to find and retrieve the trace that is generated by the current test run. Both the total length of time to wait to receive the trace, ie the 'Max wait time for trace', and the time to delay between each polling attempt, ie 'Retry delay', can be adjusted. The following diagram shows how the polling for the trace works:

Reviewer: ```suggestion
Tracetest polls the trace data store to find and retrieve the trace that is generated by the current test run. Both the total length of time to wait to receive the trace, i.e., the 'Max wait time for trace' and the time to delay between each polling attempt, i.e., 'Retry delay', can be adjusted. The following diagram shows how the polling for the trace works:
```

```mermaid
flowchart TD
    A[Run] --> B[Created]
    B --> C[Resolve Trigger Vars]
    C --> D[Execute Trigger]
    D --> ET{Is Successful Trigger}
    ET -->|Yes| E[Queue Polling]
    ET -->|NO| ES[Set State to Failed]
    ES --> Q
    E --> F[Execute Polling Job]
    F --> G[Fetch Trace from Data Store<br>after the retry delay]
    G --> H{Trace Exists}
    H -->|No| I{Max wait time<br>for trace reached}
    H -->|Yes| J{Has the span # changed}
    J -->|Yes| G
    J -->|No| K[Trace is ready]
    I -->|No| G
    I -->|Yes| L[Trace fetch failed]
    K --> O[Generating Outputs]
    O --> P[Running Test Specs]
    P --> Q[Finish]
    L --> ES
```
  

## Changing Trace Polling Settings from the UI

> In the Web UI, open settings, and select Trace Polling.

Reviewer: ```suggestion
In the Tracetest Web UI, open Settings and select Trace Polling:
```

![Demo Settings](./img/trace-polling.png)

> From this trace polling page under settings you can adjust the speed that the trace data store is polled when gathering the trace associated with a test run.

Reviewer: ```suggestion
From this trace polling page, under settings, you can adjust the speed with which the trace data store is polled when gathering the trace associated with a test run.
```

## Changing Trace Polling Settings with the CLI

> Or, if you prefer using the CLI, you can use this resource definition to set the polling parameters for the Tracetest server:

Reviewer: ```suggestion
Or, if you prefer using the CLI, you can use this resource definition file to set the polling parameters for the Tracetest server:
```

```yaml
type: PollingProfile
spec:
  default: true

>   id: current
>   name: default

Reviewer 2: same for this, not neccesary
```suggestion
```

  periodic:
    retryDelay: 500ms
    timeout: 50s
  strategy: periodic
```

> Proceed to run this command in the terminal, and specify the file above.

Reviewer: ```suggestion
Proceed to run this command in the terminal and specify the file above.
```
